addOns/regextester/src/main/java/org/zaproxy/zap/extension/regextester/RegexTester.java

```diff
@@ -0,0 +1,99 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.regextester;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.parosproxy.paros.Constant;
+
+public final class RegexTester {
+
+    private RegexTester() {}
+
+    public static RegexTestResult test(RegexTestInput input) {
+        try {
+            return testInternal(input);
+        } catch (Exception e) {
+            return new RegexTestResult(false, false, e.getMessage(), "");
+        }
+    }
+
+    private static RegexTestResult testInternal(RegexTestInput input) {
+        Pattern compiledPattern = Pattern.compile(input.getPattern());
+        Matcher matcher = compiledPattern.matcher(input.getText());
+        boolean match = matcher.matches();
+        boolean lookingAt = matcher.lookingAt();
+
+        List<RegexTestResult.MatchPositionHighlight> highlights = new ArrayList<>();
+        Matcher findMatcher = compiledPattern.matcher(input.getText());
+        StringBuilder capture = new StringBuilder();
+        int findIndex = 1;
+        boolean matcherFound = false;
+        while (findMatcher.find()) {
+            highlights.add(
+                    new RegexTestResult.MatchPositionHighlight(
+                            findMatcher.start(), findMatcher.end()));
+            appendFind(findMatcher, findIndex, capture);
+            appendAllGroups(findMatcher, findIndex, capture);
+            capture.append("\n");
+            findIndex++;
+            matcherFound = true;
+        }
+
+        if (!matcherFound) {
+            return new RegexTestResult(
+                    match, lookingAt, Constant.messages.getString("regextester.result.none"), "");
+        }
+
+        return new RegexTestResult(
+                match, lookingAt, input.getText(), capture.toString().trim(), highlights);
+    }
+
+    private static void appendFind(Matcher matcher, int findIndex, StringBuilder capture) {
+        capture.append(findIndex)
+                .append("   ")
+                .append(buildGroup(matcher.start(), matcher.end()))
+                .append("\t")
+                .append(matcher.group())
+                .append("\n");
+    }
+
+    private static void appendAllGroups(Matcher matcher, int findIndex, StringBuilder capture) {
+        for (int i = 1; i <= matcher.groupCount(); i++) {
+            capture.append("")
+                    .append(findIndex)
+                    .append(".")
+                    .append(i)
+                    .append(" ")
+                    .append(buildGroup(matcher.start(i), matcher.end(i)))
+                    .append("\t")
```

Comment:
This and following can be moved to the method as well, we just need to pass the `matcher` and the index (for `appendFind` we use index 0).